### ISSUE TITLE
Using with fanout exchange

### ISSUE THREAD

**Issue by doughsay** [conduitframework/conduit_amqp#1]
Hi, I'm trying to use conduit with simple fanout exchanges in rabbitmq.

This line: https://github.com/conduitframework/conduit_amqp/blob/master/lib/conduit_amqp/topology.ex#L38

Always binds the defined queues to the exchange with a routing key.  In fact, it requires a routing key to be defined before it even attempts to bind.  But fanout exchanges ignore routing keys (according to what I've read).

So I have a broker like this:

```elixir
configure do
  exchange "example.exchange", type: :fanout, durable: true

  queue "my.queue", exchange: "example.exchange", from: "ignored", durable: true
end
```

I have to put something in the `:from` key in the config, even though it will be ignored.

I feel like it should be possible to not specify that option, and have the binding happen anyway for fanout exchanges, but I can't think of a clean way to make that happen in this repo.

Thoughts?

**Comment by blatyo** [comment]
Yea, that was an oversight since I almost exclusively use topic exchanges. I just recently used a fanout exchange and ran into the same issue when I was building a phoenix pubsub adapter to use conduit. So, I'm also specifying a routing key.

I think it should just look for the presence of the `:from` key and if it's not present, do a bind with just the opts provided.

**Comment by blatyo** [comment]
Fixed in v0.4.5.
